### PR TITLE
:sparkles: Add options to configure a logger for manager and controllers

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

/assign @alvaroaleman @detiber 

Made these changes also backward compatible to be able to go in v0.6.2

### CODE REVIEW

File: pkg/builder/controller.go [hunk 1]
```diff
@@ -228,26 +237,33 @@ func (blder *Builder) loadRestConfig() {
...
+	gvk, err := getGvk(blder.forInput.object, blder.mgr.GetScheme())
+	if err != nil {
+		return err
+	}
+
+	// Setup the logger.
+	if ctrlOptions.Log == nil {
+		ctrlOptions.Log = blder.mgr.GetLogger()
+	}
+	ctrlOptions.Log = ctrlOptions.Log.WithValues("groupVersionKind", gvk.String())
```
Comment (alvaroaleman) [inline pkg/builder/controller.go:264]: How do we properly distinguish this from what we do in the encoder https://github.com/kubernetes-sigs/controller-runtime/blob/6633846423ba4853b1c4466f28aa6b66d6804930/pkg/log/zap/kube_helpers.go#L69 ?

I think the log key should maybe be `reoncilerGroupVersionKind` or something like that?

File: pkg/internal/controller/controller.go [hunk 2]
```diff
@@ -256,7 +258,7 @@ func (c *Controller) reconcileHandler(obj interface{}) bool {
 	c.Queue.Forget(obj)
 
 	// TODO(directxman12): What does 1 mean?  Do we want level constants?  Do we want levels at all?
-	c.Log.V(1).Info("Successfully Reconciled", "name", req.Name, "namespace", req.Namespace)
+	log.V(1).Info("Successfully Reconciled", "name", req.Name, "namespace", req.Namespace)
```
Comment (detiber) [inline pkg/internal/controller/controller.go:261]:
```suggestion
	log.V(1).Info("Successfully Reconciled")
```

File: pkg/builder/controller.go (hunk 1 above)
Comment (detiber) [reply pkg/builder/controller.go:264]: It seems like the logging here should remove the need to enrich the log entries in the zap helper. Would you expect any delta between the gvk defined here and the resource passed to reconcile?

Comment (alvaroaleman) [reply pkg/builder/controller.go:264]: Its totally possible that during the reconciliation of an object type a an object type b has to be updated and that fails so I think the two are complementary.

Comment (vincepri) [reply pkg/builder/controller.go]: Fixed, used `reconcilerGroup` and `reconcilerKind`

**Pull request merged**